src/components/pill-selector-dropdown/Pill.js

```diff
@@ -3,19 +3,29 @@ import React from 'react';
 import classNames from 'classnames';
 
 type Props = {
+    isDisabled?: boolean,
     isSelected?: boolean,
     isValid?: boolean,
     onRemove: Function,
     text: string,
 };
 
-const Pill = ({ isSelected = false, isValid = true, onRemove, text }: Props) => (
-    <span className={classNames('pill', { 'is-selected': isSelected, 'is-invalid': !isValid })}>
-        <span className="pill-text">{text}</span>
-        <span aria-hidden="true" className="close-btn" onClick={onRemove}>
-            ✕
+const Pill = ({ isDisabled = false, isSelected = false, isValid = true, onRemove, text }: Props) => {
+    const removeFn = isDisabled ? undefined : onRemove;
+    return (
+        <span
+            className={classNames('pill', {
+                'is-selected': isSelected,
```

Comment:
Can a pill be both disabled and selected at the same time or should they be mutually-exclusive?

Reply:
It looks like `is-selected` is used for keyboard accessibility of the pills, so if you for example hit the left arrow it will select the previous pill. However, when the PillSelector is disabled, these keyboard interactions won't work because the whole textarea is disabled, so by nature of the UI they seem to be mutually exclusive. I'm not sure we gain much by adding that complexity into the code though. It will not set `is-selected` if it's disabled because there is no way to select.

Reply:
I see. I think each component should make sure its internal state (in this case expressed via CSS classes) is consistent and correct, though. I think we basically just need to add `'is-selected': isSelected && !isDisabled,`, which I'm not sure adds much in the way of complexity. It's up to you, though.